runtime/service/src/main/java/org/apache/polaris/service/auth/JWTBroker.java

```diff
@@ -60,34 +58,22 @@ public abstract class JWTBroker implements TokenBroker {
   public abstract Algorithm getAlgorithm();
 
   @Override
-  public DecodedToken verify(String token) {
+  public PolarisCredential verify(String token) {
+    return verifyInternal(token);
+  }
+
+  private InternalPolarisToken verifyInternal(String token) {
     JWTVerifier verifier = JWT.require(getAlgorithm()).withClaim(CLAIM_KEY_ACTIVE, true).build();
 
     try {
       DecodedJWT decodedJWT = verifier.verify(token);
-      return new DecodedToken() {
-        @Override
-        public Long getPrincipalId() {
-          return decodedJWT.getClaim("principalId").asLong();
-        }
-
-        @Override
-        public String getClientId() {
-          return decodedJWT.getClaim("client_id").asString();
-        }
-
-        @Override
-        public String getSub() {
-          return decodedJWT.getSubject();
-        }
-
-        @Override
-        public String getScope() {
-          return decodedJWT.getClaim("scope").asString();
-        }
-      };
-
-    } catch (JWTVerificationException e) {
+      return InternalPolarisToken.of(
+          decodedJWT.getSubject(),
```

Comment:
I guess this might have issues during rolling upgrades as old tokens might still have `sub` == principal ID.

I do not think it is a critical issue, but might be worth mentioning in `CHANGELOG.md`. WDYT?

Reply:
So, if a token generated by an old `JWTBroker` is processed by a new `JWTBroker`, `getPrincipalName()` would return the string representation of the principal ID which would result in a failed lookup, but `getPrincipalId()` would still return the principal ID in numeric form, and that lookup would succeed.

Reply:
Good point! Thanks for the clarification :+1: